### PR TITLE
paste multiple paragraphs

### PR DESCRIPTION
**Pull request by yoshokatana**
* paste multiple paragraphs
* if there's already text and you paste a single paragraph, it'll append to the text
* if there are more paragraphs, it'll create them
* it runs all pasted text through `text-model`
* paragraphs are created in parallel, for speed (⌐■_■)
* after paragraphs are created, it focuses on the last one and sets the caret to the end
* it's not instant, but it's pretty fast

### CODE REVIEW

File: behaviors/wysiwyg.js [hunk 1]
```diff
@@ -13,6 +13,69 @@ var _ = require('lodash'),
...
+  // that hem hath holpen, when that they were pasted
+}
+
+/**
+ * split innerHTML into paragraphs
+ * @param {string} str
+ * @returns {array}
+ */
+function splitParagraphs(str) {
+  return str.split('¶');
```
Comment (cruzanmo) [inline behaviors/wysiwyg.js]: A far-edge case, but what happens if editorial uses the ¶ character?

Comment (yoshokatana) [inline behaviors/wysiwyg.js]: If they type that character, everything's fine. If they paste in html that contains that character, it'll make a new paragraph. Hmmmmm

Comment (yoshokatana) [inline behaviors/wysiwyg.js]: I'm gonna set it to use `'¶¶'` instead.

Comment (yoshokatana) [inline behaviors/wysiwyg.js]: actually, splitting on the closing paragraph tag gives us the best support for google docs, pages, and word, and doesn't require weird characters

File: behaviors/wysiwyg.js [hunk 2]
```diff
@@ -270,6 +333,47 @@ function addComponent(el, text) {
...
+
+  // create the new components
+  return Promise.all(_.map(components, function (component) {
+    var throwawayDiv = document.createElement('div'),
+      fragment = model.toElement(component);
+
+    throwawayDiv.appendChild(fragment);
+
+    return edit.createComponent(current.name, {
+      text: throwawayDiv.innerHTML
```
Comment (byronhulcher) [inline behaviors/wysiwyg.js]: I think you can use `fragment.outerHTML` and dont need to create a random div

File: behaviors/wysiwyg.js [hunk 3]
```diff
@@ -13,6 +13,71 @@ var _ = require('lodash'),
...
+  var range = document.createRange(),
+    selection = window.getSelection();
+
+  range.setStartAfter(node); // set the caret after this node
+  range.collapse(true);
+  selection.removeAllRanges();
+  selection.addRange(range);
+  // Of the DOM, to inputte's end they wende
+  // the hooly blisful caret for to seke
+  // that hem hath holpen, when that they were pasted
```
Comment (byronhulcher) [inline behaviors/wysiwyg.js]: Is this new es6 syntax? I'm having trouble reading this :eyes:

Comment (yoshokatana) [inline behaviors/wysiwyg.js]: ES1475 actually

File: behaviors/wysiwyg.js (hunk 2 above)
Comment (yoshokatana) [inline behaviors/wysiwyg.js]: > If the element has no parent node, that is if it is the root node of the document, setting its outerHTML property will throw a DOMException with the error code NO_MODIFICATION_ALLOWED_ERR.

Yeah, it's not great, but apparently the temp div is the current best practice :disappointed:

**Comment by byronhulcher** [comment]
:+1: to me!

**Comment by cruzanmo** [comment]
+1

**Pull request merged**